typescript/ts-node-lowering/src/org/jetrbains/dukat/nodeLowering/lowerings/convertStatements.kt

```diff
@@ -145,23 +150,103 @@ internal class ExpressionConverter(val documentConverter: DocumentConverter) {
                 }
             }
-            else -> raiseConcern("unable to process ExpressionDeclaration ${this}") {
+            else -> raiseConcern("unable to process ExpressionDeclaration $this") {
                 IdentifierExpressionModel(IdentifierEntity("ERROR"))
             }
         }
     }
 
+    private fun BlockDeclaration.isFallthroughBlock(): Boolean {
+        return when (statements.lastOrNull()) {
+            null -> true
+            is BreakStatementDeclaration -> false
+            else -> true
+        }
+    }
+
+    private fun SwitchStatementDeclaration.splitToFallthroughBlocks(): List<List<CaseDeclaration>> {
+        val fallthroughBlocks = mutableListOf<List<CaseDeclaration>>()
+        var currentBlock = mutableListOf<CaseDeclaration>()
+        cases.forEach {
+            currentBlock.add(it)
+            if (!it.body.isFallthroughBlock()) {
+                fallthroughBlocks += currentBlock
+                currentBlock = mutableListOf()
+            }
+        }
+        if (currentBlock.isNotEmpty()) {
+            fallthroughBlocks += currentBlock
+        }
+        return fallthroughBlocks
+    }
+
+    private fun StatementModel.removeBreaks(): StatementModel? {
+        return when (this) {
+            is BlockStatementModel -> removeBreaks()
+            is BreakStatementModel -> null
+            is IfStatementModel -> copy(
+                thenStatement = thenStatement.removeBreaks(),
+                elseStatement = elseStatement?.removeBreaks()
+            )
+            is RunBlockStatementModel -> copy(statements = statements.mapNotNull { it.removeBreaks() })
+            is WhenStatementModel -> copy(cases = cases.map { it.copy(body = it.body.removeBreaks()) })
+            is WhileStatementModel -> copy(body = body.removeBreaks())
+            else -> this
+        }
+    }
+
+    private fun BlockStatementModel.removeBreaks() = copy(statements = statements.mapNotNull { it.removeBreaks() })
+
+    private fun List<CaseDeclaration>.convert(expressionToCompare: ExpressionModel): CaseModel? {
+        return when (this.size) {
+            0 -> null
+            1 -> CaseModel(
+                first().condition?.convert()?.let { listOf(it) },
+                convertBlock(first().body)
+            )
+            else -> {
+                val condition = if (none { it.condition == null }) {
+                    map { it.condition!!.convert() }
+                } else {
+                    null
+                }
+                val body = map { case ->
+                    if (case.condition == null) {
+                        case.body.statements.map { it.convert() }
+                    } else {
+                        listOf(
+                            IfStatementModel(
+                                BinaryExpressionModel(
+                                    expressionToCompare,
+                                    "==",
```

Comment:
We should finally narrow BinaryExpressionModel::operator but it wouldn't be honest from my side to ask you to do it in this particular PR.  However what makes sense  even in the framework of this PR is to move this operators to some constants. Can you please do it?

Reply:
Fixed - now binary and unary operators are enums in kotlin model